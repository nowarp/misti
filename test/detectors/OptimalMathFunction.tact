const ONE: Int = 1;
const TWO: Int = ONE + 1;

fun toCall(): Int {
    return 42;
}

fun test1() {
    // Suggest pow2:
    pow(3, 2);
    pow(3, 1 + 1);

    // OK: consteval does not support constants yet
    pow(3, TWO);
    pow(3, ONE + ONE);

    // OK: ensure consteval doesn't crash
    pow(toCall(), toCall());
    pow(toCall(), toCall() + 1);
}
