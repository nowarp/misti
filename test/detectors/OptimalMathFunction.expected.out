[LOW] OptimalMathFunction: Use more gas-efficient function: pow2(3)
test/detectors/OptimalMathFunction.tact:10:5:
   9 |     // Suggest pow2:
> 10 |     pow(3, 2);
           ^
  11 |     pow(3, 1 + 1);
Help: Choose more gas-efficient function
See: https://nowarp.io/tools/misti/docs/detectors/OptimalMathFunction

[LOW] OptimalMathFunction: Use more gas-efficient function: pow2(3)
test/detectors/OptimalMathFunction.tact:11:5:
  10 |     pow(3, 2);
> 11 |     pow(3, 1 + 1);
           ^
  12 | 
Help: Choose more gas-efficient function
See: https://nowarp.io/tools/misti/docs/detectors/OptimalMathFunction